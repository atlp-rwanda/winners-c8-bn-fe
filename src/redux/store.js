// this file will configure the redux's store
import { createStore } from "redux";
import rootReducer from "./reducers";

let authToken = ""; 
if(window.localStorage.getItem("auth-token")) {
  authToken = window.localStorage.getItem("auth-token");
}

const initialState = {
    auth: {token: authToken},
    navbar: {currentPage: "HOME"},
};
const store = createStore(
    rootReducer,
    initialState // sets initial state
  );

export default store;
