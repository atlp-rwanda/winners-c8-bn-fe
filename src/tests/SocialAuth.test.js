import React from 'react'
import { Provider } from 'react-redux';
import  SocialAuth  from '../components/SocialOAuth';
import renderer from 'react-test-renderer'
import { MemoryRouter} from 'react-router-dom';
import store from '../redux/store'
// import {render} from '@testing-library/jest-dom'


test('<SocialAuth /> matches snapshot', () => {
    const component = renderer
		.create(
			<Provider store={store}>
				<MemoryRouter>
					<SocialAuth />
				</MemoryRouter>
			</Provider>,
		)
		.toJSON();
	expect(component).toMatchSnapshot();
});
