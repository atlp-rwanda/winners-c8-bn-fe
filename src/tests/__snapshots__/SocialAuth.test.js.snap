// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialAuth /> matches snapshot 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginTop": "50px",
    }
  }
>
  <div
    style={
      Object {
        "paddingRight": "40px",
      }
    }
  >
    <a
      href="https://winners-c8-bn-be-staging.herokuapp.com/api/oauth/google"
    >
      <svg
        enableBackground="new 0 0 48 48"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
            "cursor": "pointer",
            "fontSize": "40px",
          }
        }
        version="1.1"
        viewBox="0 0 48 48"
        width="1em"
        x="0px"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <path
          d="M43.611,20.083H42V20H24v8h11.303c-1.649,4.657-6.08,8-11.303,8c-6.627,0-12-5.373-12-12
	c0-6.627,5.373-12,12-12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C12.955,4,4,12.955,4,24
	c0,11.045,8.955,20,20,20c11.045,0,20-8.955,20-20C44,22.659,43.862,21.35,43.611,20.083z"
          fill="#FFC107"
        />
        <path
          d="M6.306,14.691l6.571,4.819C14.655,15.108,18.961,12,24,12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657
	C34.046,6.053,29.268,4,24,4C16.318,4,9.656,8.337,6.306,14.691z"
          fill="#FF3D00"
        />
        <path
          d="M24,44c5.166,0,9.86-1.977,13.409-5.192l-6.19-5.238C29.211,35.091,26.715,36,24,36
	c-5.202,0-9.619-3.317-11.283-7.946l-6.522,5.025C9.505,39.556,16.227,44,24,44z"
          fill="#4CAF50"
        />
        <path
          d="M43.611,20.083H42V20H24v8h11.303c-0.792,2.237-2.231,4.166-4.087,5.571
	c0.001-0.001,0.002-0.001,0.003-0.002l6.19,5.238C36.971,39.205,44,34,44,24C44,22.659,43.862,21.35,43.611,20.083z"
          fill="#1976D2"
        />
      </svg>
    </a>
  </div>
  <div
    style={
      Object {
        "paddingLeft": "30px",
      }
    }
  >
    <a
      href="https://winners-c8-bn-be-staging.herokuapp.com/api/oauth/facebook"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "1D3C78",
            "cursor": "pointer",
            "fontSize": "40px",
          }
        }
        version="1.1"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.5 3h2.5v-3h-2.5c-1.93 0-3.5 1.57-3.5 3.5v1.5h-2v3h2v8h3v-8h2.5l0.5-3h-3v-1.5c0-0.271 0.229-0.5 0.5-0.5z"
        />
      </svg>
    </a>
  </div>
</div>
`;
